### PR TITLE
fix: Add retry logic to add apt repository tasks

### PR DESCRIPTION
**Pull request by nadeemshahzad**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

File: playbooks/roles/common/tasks/main.yml [hunk 1]
```diff
@@ -100,6 +105,11 @@
 - name: add deadsnakes repository
   apt_repository:
     repo: "ppa:deadsnakes/ppa"
+    update_cache: yes
```
Comment (adzuci) [inline playbooks/roles/common/tasks/main.yml:113]: Can you also add this here? https://github.com/edx/configuration/pull/6506/files#diff-6b9086567f8ddca7d54ce7bf6b8e89b7974a1b65edefb76f3bd2087d281c4b7eR71

File: playbooks/roles/common/tasks/main.yml [hunk 2]
```diff
@@ -59,6 +59,11 @@
 - name: Add git apt repository
   apt_repository:
     repo: "{{ common_git_ppa }}"
+    update_cache: yes
```
Comment (adzuci) [inline playbooks/roles/common/tasks/main.yml:62]: This will be pretty slow right?  Do you know if this was happening before?

Comment (NIXKnight) [reply playbooks/roles/common/tasks/main.yml:62]: According to the module docs by default `update_cache` is set to `yes`. I believe that adding the parameter explicitly will cause no harm.

Comment (adzuci) [reply playbooks/roles/common/tasks/main.yml:62]: Nvm, this was already yes and you are just making explicit

**Review by adzuci — APPROVED** [review]
:shipit:

File: playbooks/roles/common/tasks/main.yml (hunk 2 above)
Comment (matthugs) [reply playbooks/roles/common/tasks/main.yml:62]: I'm not sure that's true.

Screen Shot 2021-08-12 at 1 15 10 PM
from https://docs.ansible.com/ansible/2.8/modules/apt_module.html?highlight=apt

**Pull request merged**